db-service/test/cqn4sql/column.element.test.js

```diff
@@ -162,4 +162,16 @@ describe('assign element onto columns with flat model', () => {
     // foreign key is part of flat model
     expect(query.SELECT.columns[1]).to.have.property('element').that.eqls(TestPublisher.elements.publisherRenamedKey_notID)
   })
+
+  it('attaches foreign key and not the association as element', () => {
+    let query = cqn4sql(cds.ql`
+      SELECT from bookshop.Books as Books {
+        genre.parent.ID,
+        genre.parent as assoc
+      }
+    `, model)
+    const { Genres } = model.entities
+    expect(query.SELECT.columns[0]).to.have.property('element').that.eqls(Genres.elements.ID)
```

Comment:
could also be `Genres:parent_ID` but the path navigates to the target → fine IMO. Semantically this should not make any difference.